lib/dtutils.lua

```diff
@@ -0,0 +1,457 @@
+  else
+    log.msg(log.info, "Error loading " .. req_name)
+  end
+  return lib
+end
+
+--[[
+  NAME
+    push - push a value on a stack
+
+  SYNOPSIS
+    local du = require "lib/dtutils"
+
+    du.push(stack, value)
+      stack - table - a table being used as the stack
+      value - any type - the value to be put on the stack
+
+  DESCRIPTION
+    Push a value on a stack
+
+  RETURN VALUE
+    none
+
+]]
+
+dtutils.libdoc.functions[#dtutils.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Return_Value"},
+  Name = [[push - push a value on a stack]],
+  Synopsis = [[local du = require "lib/dtutils"
+
+    du.push(stack, value)
+      stack - table - a table being used as the stack
+      value - any type - the value to be put on the stack]],
+  Description = [[Push a value on a stack]],
+  Return_Value = [[none]],
+}
+
+dtutils.push = table.insert
+
+--[[
+  NAME
+    pop - pop a value from a stack
+
+  SYNOPSIS
+    local du = require "lib/dtutils"
+
+    local result = du.pop(stack)
+      stack - table - a table being used as a stack
+
+  DESCRIPTION
+    Remove the last value pushed on the stack and return it
+
+  RETURN VALUE
+    result = nil if stack isn't a table or the stack is empty, otherwise the value removed from the stack
+
+]]
+
+dtutils.libdoc.functions[#dtutils.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Return_Value"},
+  Name = [[pop - pop a value from a stack]],
+  Synopsis = [[local du = require "lib/dtutils"
+
+    local result = du.pop(stack)
+      stack - table - a table being used as a stack]],
+  Description = [[Remove the last value pushed on the stack and return it]],
+  Return_Value = [[result = nil if stack isn't a table or the stack is empty, otherwise the value removed from the stack]],
+}
+
+dtutils.pop = table.remove
+
+--[[
+  NAME
+    update_combobox_choices - change the list of choices in a combobox
+
+  SYNOPSIS
+    local du = require "lib/dtutils"
+
+    du.update_combobox_choices(combobox_widget, choice_table)
+      combobox_widget - lua_combobox - a combobox widget
+      choice_table - table - a table of strings for the combobox choices
+
+  DESCRIPTION
+    Set the combobox choices to the supplied list.  Remove any extra choices from the end
+
+  RETURN VALUE
+    none
+
+]]
+
+dtutils.libdoc.functions[#dtutils.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Return_Value"},
+  Name = [[update_combobox_choices - change the list of choices in a combobox]],
+  Synopsis = [[local du = require "lib/dtutils"
+
+    du.update_combobox_choices(combobox_widget, choice_table)
+      combobox_widget - lua_combobox - a combobox widget
+      choice_table - table - a table of strings for the combobox choices]],
+  Description = [[Set the combobox choices to the supplied list.  Remove any extra choices from the end]],
+  Return_Value = [[none]],
+}
+
+function dtutils.update_combobox_choices(combobox, choice_table)
```

Comment:
sounds good

I have thought about that one, and I understand why it makes me uneasy...

this is a "replace in place" function for table, i.e it takes two tables and set the first table to hav e the same elements than the second table

this is disturbing because it's something you should never have to do in lua, but the widget API (and maybe other API) need it.

This function probably can stay in dtutils, but with a different name. dtutils.replace_table_elements or something similar...